core/cont/src/TClassTable.cxx

```diff
@@ -590,15 +649,22 @@ DictFuncPtr_t TClassTable::GetDict(const char *cname)
 
 DictFuncPtr_t TClassTable::GetDict(const std::type_info& info)
 {
-   if (!CheckClassTableInit()) return nullptr;
+   if (!CheckClassTableInit())
+      return nullptr;
+
+   if (gDebug > 9)
+      gROOT; // Info migth recursively call TClassTable during the gROOT init
+
+   std::lock_guard<std::mutex> lock(GetClassTableMutex());
 
    if (gDebug > 9) {
       ::Info("GetDict", "searches for %s at 0x%zx", info.name(), (size_t)&info);
```

Comment:
Does this suffer from the recursive lock issue, too?

Reply:
Line 656 is intended to prevent it from happening by forcing the initialization of `gROOT`.